tsdb/db.go

```diff
@@ -1642,6 +1642,11 @@ func rangeForTimestamp(t int64, width int64) (maxt int64) {
 	return (t/width)*width + width
 }
 
+func (db *DB) overlapsInterval(mint, maxt int64) bool {
+	return (db.head.MinTime() >= mint && db.head.MinTime() <= maxt) ||
+		(db.head.MaxTime() >= mint && db.head.MaxTime() <= maxt)
+}
```

Comment:
Shouldn't this be part of the `Head` struct? And you will find a simpler `overlapsClosedInterval` there :)